### DISCUSSION TITLE
Push/callback notifications

### DISCUSSION THREAD

**Discussion by eeker01** [weather-gov/api#741] [Ideas]
Are there any plans to implement some sort of a "push" (or callback) approach?

I'm thinking, an app registers a callback URL, and when the API has something new for the client to grab, it hits all of the callbacks that have valid registrations for the specific endpoint, and perhaps location polygon (primarily, I'm thinking about watch and warning types of alerts, where the timing and location may be critical). Just a random thought. I'm still getting my feet wet with this API - so far, I like what I see.

**Comment by dwhitemv25** [comment]
The current push service is [NWS Weather Wire](https://www.weather.gov/nwws/). NWWS sends the text products themselves so there's some extra parsing involved, but it's no cost to get on Open Interface, and there's 3rd party example clients that can be extended to do whatever you want. You could build your own push-based service on top of it.

API in its current form is a pull model and is reliant on caching, but I wonder if Akamai has something that can help scale push notifications.

**Reply by eeker01** [reply]
Oh, nice. I'll have to check it out.

Thanks!
